aten/src/TH/generic/THVectorDispatch.cpp

```diff
@@ -237,4 +239,6 @@ struct THVector_(startup) {
 // Declare a global instance to force static initialization
 static THVector_(startup) THVector_(g_startup);
 
+#endif /* non bool only part */
```

Comment:
same comment as the .h.

Reply:
I though about this but i'm not sure if that's the best approach:
1) THVectorDispatch.cpp
It has no header file.
We need THVector_(fill) which requires THVector_(fill_DISPATCHPTR), both are in this file. The problem with moving them out is in THVector_(startup) structure (same file) which initializes all the methods. I agree that having special checks for bool is not good but breaking the whole thing apart is worse as it breaks apart the THVector related methods.

2) THTensorMath.h
Similar story here. We can move the _(fill) thing out but than there is a question what to do with _(zero) that uses it? If we move it, we will end up with a file that has only 2 methods that are somewhat related to each other? Not ideal. My reasoning is that TH will be moved to Aten eventually and it should be cleaned up. 

That's one of the reasons why i rolled back my PR with refactoring stuff to TensorAssignments, it looked messy.

Reply:
In case, if we decide to go ahead and move fill() and zero() from THTensorMath, here is a [PR](https://github.com/pytorch/pytorch/pull/17536)

Reply:
I don't think Greg is asking you to move these functions; I think he's referring to "can't you end the define at line 24?" on THVector.h

Reply:
oh i see. No, not really. There are a bunch of methods that we dont need for bool but need for all other types, so its ether this way or we have to specify all supported types for _fill() method including bool. Not sure which solution is more clear.